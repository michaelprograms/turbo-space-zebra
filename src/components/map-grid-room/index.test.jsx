import { expect, test } from 'vitest';
import { render } from '@testing-library/react';
// import { find } from 'styled-components/test-utils'

import MapGridRoom from './';

describe('<MapGridRoom />', () => {
  test('MapGridRoom mounts properly', () => {
    const wrapper = render(<MapGridRoom />);
    expect(wrapper).toBeTruthy();
  });

  test('MapGridRoom displays room symbol', () => {

    // No Symbol and No Exits
    let wrapper = render(<MapGridRoom />);
    expect(wrapper.container).toMatchSnapshot();
    expect(wrapper.container.children[0].children.length).toBe(0);

    // Symbol and No Exits
    wrapper = render(<MapGridRoom room={{ enabled: true }} />);
    expect(wrapper.container).toMatchSnapshot();
    expect(wrapper.container.children[0].children.length).toBe(1);

    // No Symbol and Exits
    wrapper = render(<MapGridRoom room={{
      enabled: false,
      northEnabled: true,
      northeastEnabled: true,
      eastEnabled: true,
      southeastEnabled: true,
      southEnabled: true,
      southwestEnabled: true,
      westEnabled: true,
      northwestEnabled: true,
    }} />);
    expect(wrapper.container).toMatchSnapshot();
    expect(wrapper.container.children[0].children.length).toBe(8);

    // Symbol and Exits
    wrapper = render(<MapGridRoom room={{
      enabled: true,
      northEnabled: true,
      northeastEnabled: true,
      eastEnabled: true,
      southeastEnabled: true,
      southEnabled: true,
      southwestEnabled: true,
      westEnabled: true,
      northwestEnabled: true,
    }} />);
    expect(wrapper.container).toMatchSnapshot();
    expect(wrapper.container.children[0].children.length).toBe(9);
  });
});