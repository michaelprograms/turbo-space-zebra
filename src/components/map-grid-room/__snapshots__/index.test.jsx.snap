// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MapGridRoom /> > MapGridRoom displays room symbol 1`] = `
<div>
  <div
    class="sc-beySPh eZCKBQ"
  />
</div>
`;

exports[`<MapGridRoom /> > MapGridRoom displays room symbol 2`] = `
<div>
  <div
    class="sc-beySPh eZCKBQ"
  >
    <div
      class="sc-guDLey LVYym"
    />
  </div>
</div>
`;

exports[`<MapGridRoom /> > MapGridRoom displays room symbol 3`] = `
<div>
  <div
    class="sc-beySPh eZCKBQ"
  >
    <div
      class="line-northwest-wrapper"
    >
      <div
        class="line line-northwest"
      />
    </div>
    <div
      class="line-northeast-wrapper"
    >
      <div
        class="line line-northeast"
      />
    </div>
    <div
      class="line-southwest-wrapper"
    >
      <div
        class="line line-southwest"
      />
    </div>
    <div
      class="line-southeast-wrapper"
    >
      <div
        class="line line-southeast"
      />
    </div>
    <div
      class="line line-north"
    />
    <div
      class="line line-south"
    />
    <div
      class="line line-west"
    />
    <div
      class="line line-east"
    />
  </div>
</div>
`;

exports[`<MapGridRoom /> > MapGridRoom displays room symbol 4`] = `
<div>
  <div
    class="sc-beySPh eZCKBQ"
  >
    <div
      class="line-northwest-wrapper"
    >
      <div
        class="line line-northwest"
      />
    </div>
    <div
      class="line-northeast-wrapper"
    >
      <div
        class="line line-northeast"
      />
    </div>
    <div
      class="line-southwest-wrapper"
    >
      <div
        class="line line-southwest"
      />
    </div>
    <div
      class="line-southeast-wrapper"
    >
      <div
        class="line line-southeast"
      />
    </div>
    <div
      class="line line-north"
    />
    <div
      class="line line-south"
    />
    <div
      class="line line-west"
    />
    <div
      class="line line-east"
    />
    <div
      class="sc-guDLey LVYym"
    />
  </div>
</div>
`;
